### ISSUE TITLE
Docs: Is Next.js a TS only platform now? All guide projects are in TS.

### ISSUE THREAD

**Issue by APNSound** [vercel/next.js#46100]
### What is the improvement or update you wish to see?

Do I have to know Type Script to use Next.js? All of the repositories referenced by official documentation point to projects written in TS. I'm just new enough to coding that this is a big deal: do I start my project over in a different framework that has JS examples? Do I keep my project at some older version of Next so I can follow along with JS examples? 

### Is there any context that might help us understand?

All GitHub repositories linked to by NextJS online docs are in Type Script. How may I view current best practices (Auth v4, for example) using JavaScript?

**Comment by Izhaki** [comment]
> Do I have to know Type Script to use Next.js?

No. When you setup your NextJS project you'll be asked whether you want it to be typescript or not.

> All of the repositories referenced by official documentation point to projects written in TS.

Not sure what "all repositories" mean here, but:

- Most [examples](https://github.com/vercel/next.js/tree/canary/examples) are written in typescript, these examples are very community driven and those writing them can choose whether to use typescript or not. It is much easier to "strip" typescript out (in your mind, or the code) then having to "fill in" the types yourself. It's very little time to go TS > JS, but JS > TS could take a lot of time (to a lot of people).
- The Next JS codebase itself takes a mixed approach:
  - In some cases, code is typescript.
  - In others, code is javascript, but with sibling typescript definitions.

> I'm just new enough to coding that this is a big deal: do I start my project over in a different framework that has JS examples?

Reading typescript (for all but very complex types) is not that hard - and if you know what is the type in `(name: string)`, then you can just ignore it in your brain.

Writing in typescript could be challenging (albeit not for trivial code), but you don't have to.

Pretty much every project I created in the last 6 years was typescript, but in many of them I didn't type a thing! Typescript infer a lot of the types, so you can see them in your IDE, then chose to type things when you feel it is needed.

Also note that you can disable typechecking in builds and IDE - so even if the code is typescript, you never get type errors breaking anything.

This is not something specific to NextJS, by the way. Typescript add a viable type-checking to an otherwise untyped language. It has many advantages, and other than very old projects, nearly all projects nowadays are typed.

**Comment by APNSound** [comment]
I don't mind learning TypeScript, I just really don't want to learn it before I can move forward. I get that it has advantages but it's frustrating to have the example on official docs lead to something I can't read. 

While I think the community supported aspect is actually a very special thing compared to other industries, it narrows access for some people to have Next make assumptions of knowledge. It's also not quite as advertised; sure you don't HAVE to use TS, but if you don't know TS, you can't understand all of the official docs, as they are a mixed bag.

Here's an example. What's in the '../lib/session'? 
https://nextjs.org/docs/authentication
(that's a rhetorical question, I don't expect you to walk me through something)

I guess I'll stop production so I can start an Udemy course on TypeScript. :/

Thanks for the time and input, I do appreciate it. I just feel frustrated, instead of a 15 minute set-up / walk through to secure my app, it's another day of researching or training. Sucks. (I have found simple walk throughs to set up NextAuth, but this tutorial above is specifically guiding server side auth. It's always the case that my needs are just complex enough to not be covered by the simple set up tutorial. This is not the first time this has happened; the code examples linked to by Next in all of their walk throughs have all been 'updated' to be TS)

**Comment by Izhaki** [comment]
Sorry if I'm being daft, but none of the examples on https://nextjs.org/docs/authentication is a typescript one.

What have I missed?

**Comment by APNSound** [comment]
"You can view this [example in action](https://iron-session-example.vercel.app/). Check out the [with-iron-session](https://github.com/vercel/next.js/tree/canary/examples/with-iron-session) example to see how it works."

with-iron-session = TS

**Comment by Izhaki** [comment]
OK. I get it now.

The only thing I can assure you now, is that the sort of types you see in these examples are very basic - there is hardly anything to learn about them (udamy course will teach you much more than you need to know for these).

This should also be more than you need, but you could read this in 1 hours: https://www.typescriptlang.org/docs/handbook/2/everyday-types.html

Slightly more advanced, but you'll see these a lot as well:
https://www.typescriptlang.org/docs/handbook/2/generics.html

Other than that, you can just refer to the manual if you don't know something.

**Comment by APNSound** [comment]
Thank you for the links, I will start there.

**Comment by icyJoseph** [comment]
@APNSound You can always navigate the repository history, and find the example when it used JavaScript, using GitHub's "History" button. 

- https://github.com/vercel/next.js/tree/84e47b82a51b3a2b0b12f1584739a6b4fa95b05f/examples/with-iron-session

Now it is not ideal because various things have changed since, for example Next.js itself, iron-session, and what not.

Either way I'd say that if you are looking for a quick reference, either the old versions in JS or newer in TS should suffice. Most of TypeScript in frontend is type annotations, if you prefer not to learn them, you can always just remove them in your editor.

**Comment by delbaoliveira** [comment]
Hey @APNSound, 

Thank you for the feedback. Making our educational assets simple and approachable for everyone is the top priority.

First, you may find this tool that takes in TypeScript code, removes types, and produces plain JavaScript useful: https://transform.tools/typescript-to-javascript

- Code snippets in future docs (beta.nextjs.org/docs) will include a code tab for TypeScript and JavaScript. Note: Code tabs are yet not implemented. 
- Create Next.js App - this tool starts JavaScript first and you have the option to add TypeScript.
- Examples (https://github.com/vercel/next.js/tree/canary/examples) - We decided to move examples to TypeScript to help improve the overall quality and consistency across releases and help steer the community towards where we believe the industry is heading. The examples themselves are usually small and isolated enough that they are an excellent way to dip your toes into TypeScript. If you can get past the initial hurdle of starter TypeScript, basic inline types aid in readability and understanding.

Overall, TypeScript's popularity is undeniable, and its usefulness is proven - we want to do our part to nudge the community in the right direction without leaving anyone behind.

**Comment by balazsorban44** [comment]
Closing as per the above comment. :pray:

**Comment by APNSound** [comment]
Thanks for the input @delbaoliveira and @icyJoseph .
No matter how you cut it, 6 months ago I was able to visit NextJS examples for a current example in JavaScript, and now I can not. I chose Next because of it's features and that it's in JavaScript; a widely supported language that I can understand as someone that started coding last year. It's fine to say I should learn TypeScript, and someday I will, but it's frustrating to be on the last steps of deployment for something and now I have to put it on hold, or delay it.

I think it should just be called NextTS and then people like me can decide earlier if we want to try and climb that hill. It's not your fault I don't know what I'm doing, but it is marketed by Vercel as a JavaScript frame work. As a one man show literally learning everything needed to create my solution, adding required languages to my skillset is killer. I'm 60+ hours  deep on training various coding aspects on Udemy already, how much more must I give to use the Next framework? 

Intended or not, making the extended examples TypeScript only is gate keeping. Of all my rant, I hope this sentiment makes it into the feedback bin.
